oit-dual-depth-peeling.html

```diff
@@ -0,0 +1,722 @@
+            console.error(gl.getShaderInfoLog(fullScreenQuadVertexShader));
+        }
+
+        var finalFsSource = document.getElementById("fragment-final").text.trim();
+        var blendBackFsSource = document.getElementById("fragment-blend-back").text.trim();
+
+        var finalFragmentShader = gl.createShader(gl.FRAGMENT_SHADER);
+        gl.shaderSource(finalFragmentShader, finalFsSource);
+        gl.compileShader(finalFragmentShader);
+
+        if (!gl.getShaderParameter(finalFragmentShader, gl.COMPILE_STATUS)) {
+            console.error(gl.getShaderInfoLog(finalFragmentShader));
+        }
+
+        var blendBackFragmentShader = gl.createShader(gl.FRAGMENT_SHADER);
+        gl.shaderSource(blendBackFragmentShader, blendBackFsSource);
+        gl.compileShader(blendBackFragmentShader);
+
+        if (!gl.getShaderParameter(blendBackFragmentShader, gl.COMPILE_STATUS)) {
+            console.error(gl.getShaderInfoLog(blendBackFragmentShader));
+        }
+
+        var finalProgram = gl.createProgram();
+        gl.attachShader(finalProgram, fullScreenQuadVertexShader);
+        gl.attachShader(finalProgram, finalFragmentShader);
+        gl.linkProgram(finalProgram);
+
+        if (!gl.getProgramParameter(finalProgram, gl.LINK_STATUS)) {
+            console.error(gl.getProgramInfoLog(finalProgram));
+        }
+
+        var blendBackProgram = gl.createProgram();
+        gl.attachShader(blendBackProgram, fullScreenQuadVertexShader);
+        gl.attachShader(blendBackProgram, blendBackFragmentShader);
+        gl.linkProgram(blendBackProgram);
+
+        if (!gl.getProgramParameter(blendBackProgram, gl.LINK_STATUS)) {
+            console.error(gl.getProgramInfoLog(blendBackProgram));
+        }
+
+        /////////////////////////
+        // GET UNIFORM LOCATIONS
+        /////////////////////////
+
+        var sceneUniformsLocation = gl.getUniformBlockIndex(dualDepthPeelingProgram, "SceneUniforms");
+        gl.uniformBlockBinding(dualDepthPeelingProgram, sceneUniformsLocation, 0);
+
+        var dualDepthPeelingLocations = {
+            texture: gl.getUniformLocation(dualDepthPeelingProgram, "uTexture"),
+            model: gl.getUniformLocation(dualDepthPeelingProgram, "uModel"),
+            depth: gl.getUniformLocation(dualDepthPeelingProgram, "uDepth"),
+            frontColor: gl.getUniformLocation(dualDepthPeelingProgram, "uFrontColor")
+        };
+
+        var finalLocations = {
+            frontColor: gl.getUniformLocation(finalProgram, "uFrontColor"),
+            backColor: gl.getUniformLocation(finalProgram, "uBackColor")
+        };
+
+        var blendBackLocations = {
+            backColor:gl.getUniformLocation(blendBackProgram, "uBackColor")
+        };
+
+        ////////////////////////////////
+        //  SET UP FRAMEBUFFERS
+        ////////////////////////////////
+
+        
+        var allBuffers = [gl.createFramebuffer(), gl.createFramebuffer()];  // 2 for ping-pong

```

Comment:
Could you think about renaming them, or adding some comments about what type of data will be stored in each one? Not sure what would be a good name, maybe `depthPeelBuffer` to match the program name?

Similar to above, I'd prefer these be separate variables, unless they're going to be in a loop. E.g.
```
var depthPeelBufferA = gl.createFramebuffer();
var depthPeelBufferB = gl.createFramebuffer();
``` 

The colorBuffer array seems ok, since it is used in a loop.

Reply:
Both of them are used in the loop. I would add some comments stating each COLOR_ATTACHMENT